<a href="https://colab.research.google.com/github/cazlima/exerciciosPython/blob/main/Carinatreinando2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
#megasena
import random



In [12]:
%%timeit
lista=[]

for n in range (10):

  zero=0
  um=0
  dois=0
  tres=0
  quatro=0
  cinco=0
  seis=0

  for i in range (2500000): 

    minha=set(random.sample(range(1,61),6))

    sorteio=set(random.sample(range(1,61),6))

    acertos=minha.intersection(sorteio)

    num_acer=len(acertos)

    if num_acer == 0:
        zero+=1
    elif num_acer == 1:
        um+=1
    elif  num_acer == 2:
        dois+=1
    elif  num_acer == 3:
        tres+=1
    elif  num_acer == 4:
        quatro+=1
    elif  num_acer == 5:
        cinco+=1
        print(i)
        lista.append(i)
        break
    else:
        seis+=1

  print(f'eu acertei um numero {um} vezes')
  print(f'eu acertei dois numeros {dois} vezes')
  print(f'eu acertei tres numeros {tres} vezes')
  print(f'eu acertei quatro numeros {quatro} vezes')
  print(f'eu acertei cinco numeros {cinco} vezes')
  print(f'eu acertei seis numeros {seis} vezes')
summ=sum(lista)
den=len(lista)
media=summ/den
print(media)


23856
eu acertei um numero 9026 vezes
eu acertei dois numeros 2228 vezes
eu acertei tres numeros 255 vezes
eu acertei quatro numeros 11 vezes
eu acertei cinco numeros 1 vezes
eu acertei seis numeros 0 vezes
116260
eu acertei um numero 44092 vezes
eu acertei dois numeros 10983 vezes
eu acertei tres numeros 1177 vezes
eu acertei quatro numeros 60 vezes
eu acertei cinco numeros 1 vezes
eu acertei seis numeros 0 vezes
12413
eu acertei um numero 4713 vezes
eu acertei dois numeros 1102 vezes
eu acertei tres numeros 128 vezes
eu acertei quatro numeros 7 vezes
eu acertei cinco numeros 1 vezes
eu acertei seis numeros 0 vezes
47159
eu acertei um numero 17840 vezes
eu acertei dois numeros 4420 vezes
eu acertei tres numeros 484 vezes
eu acertei quatro numeros 25 vezes
eu acertei cinco numeros 1 vezes
eu acertei seis numeros 0 vezes
65681
eu acertei um numero 24978 vezes
eu acertei dois numeros 6239 vezes
eu acertei tres numeros 617 vezes
eu acertei quatro numeros 31 vezes
eu acertei cinco numeros 

In [10]:
print(round(media))

178863


In [3]:
%reset

Once deleted, variables cannot be recovered. Proceed (y/[n])? y


In [9]:
%whos

Variable   Type      Data/Info
------------------------------
acertos    set       {39, 41, 50, 51, 28}
cinco      int       1
den        int       10
dois       int       7027
i          int       74245
lista      list      n=10
media      float     178863.1
minha      set       {39, 41, 50, 51, 59, 28}
n          int       9
num_acer   int       5
quatro     int       31
random     module    <module 'random' from '/u<...>lib/python3.7/random.py'>
seis       int       0
sorteio    set       {39, 41, 11, 50, 51, 28}
summ       int       1788631
tres       int       696
um         int       28008
zero       int       38483


In [13]:


lista=[]

for n in range (1000):

  zero=0
  um=0
  dois=0
  tres=0
  quatro=0
  cinco=0
  seis=0

  for i in range (2500000): 

    minha=set(random.sample(range(1,61),6))

    sorteio=set(random.sample(range(1,61),6))

    acertos=minha.intersection(sorteio)

    num_acer=len(acertos)

    if num_acer == 0:
        zero+=1
    elif num_acer == 1:
        um+=1
    elif  num_acer == 2:
        dois+=1
    elif  num_acer == 3:
        tres+=1
    elif  num_acer == 4:
        quatro+=1
    elif  num_acer == 5:
        cinco+=1
        lista.append(i)
        break
    else:
        seis+=1

summ=sum(lista)
den=len(lista)
media=round(summ/den)
print(f'Para acertar a quina preciso jogar em media {media} vezes')

Para acertar a quina preciso jogar em media 32075 vezes


esse resultado 32075 eh pq eu rodei pra n=1 antes de salvar. quando rodei pra n=1000 deiem torno de 150000vezes que preciso jogar na megasena. nao rodo novamente pq levou uns bons 40 minutos :)

In [14]:
%reset


Once deleted, variables cannot be recovered. Proceed (y/[n])? y
